src/data/clean_data.py

```diff
@@ -0,0 +1,245 @@
+        _awesome_cossim_top then finds the similarity between column values, returning a df, then all exact matches are removed.
+        Finally replacing the original values of the df.
+        """
+        unique_names = self.df[self.column_name].unique()
+        #print(len(unique_names))  # Uncomment to compare how many values have been combined
+        vectorizer = TfidfVectorizer(min_df=1, analyzer=self._ngrams)
+        tf_idf_matrix = vectorizer.fit_transform(unique_names)
+        matches = self._awesome_cossim_top(tf_idf_matrix, tf_idf_matrix.transpose(), 100)
+        matches_df = self._get_matches_df(matches, unique_names)
+        matches_df = matches_df[matches_df['similairity'] < 0.99999] # Remove all exact matches
+        # Future improvement: Use the highest value count of either left or right side to determine which to use as final value.
+        for left_side, right_side in zip(matches_df['left_side'], matches_df['right_side']):
+            for position, column in zip(self.df.index, self.df[self.column_name]):
+                if column == right_side:
+                    self.df.at[position, self.column_name] = left_side
+        #print(len(self.df[self.column_name].unique())) # Uncomment to compare how many values have been combined
+        return self.df
+
+
+def build_classes(path: str, lowest_similarity: float, ngram_size: int):
+    """
+    Takes a csv file and combines similar values using ngram_size to determine string chunk sizing.
+    Use return_df_as_csv(build_classes(path: str, lowest_similarity: float, ngram_size: int), filename: str)
+        to output a csv file.
+
+    Parameters:
+        path: str
+            Path to the CSV file to be cleaned
+        lowest_similarity: float
+            Lowest similarity percentage of column values to combine.
+                i.e. 0.9 = words with 90% or more similarity are combined.
+        ngram_size: int
+            Size of string chunks used to assess similarity between two values.
+            3 is normally best but values between 2 and 5 can work.
+
+    Returns:
+        A Pandas Dataframe with all columns' values with similarity at or above lowest_similarity combined.
+
+    """
+    df = _csv_to_df(path)
+    column_list = _get_df_columns(df)
+    count = 0
+    name_list = []
+    for i in range(len(column_list)+1):
+        name_list.append(f"cleaner{str(i)}")   
+    replace = DataCleaner._replace_matches_df
+    for column in column_list:
+        print("Cleaning " + column + " column " + "of " + path)
+        if count < 1:
+            name_list[count] = DataCleaner(df, lowest_similarity, column, ngram_size)
+            count += 1
+            name_list[count-1]
+        
+        name_list[count] = DataCleaner(replace(name_list[count-1]), lowest_similarity, column, ngram_size)
+        count += 1
+        name_list[count-1]
+    return replace(name_list[count-1])  
+
+def return_df_as_csv(df, filename: str):
```

Comment:
No return statement in this function, I think `write_df_as_csv()` might be more appropriate

Reply:
Honestly this makes so much sense and yet the pain of writing the `build_class` function (getting renamed) fried my brain. I went through a bunch of ideas of making the `DataCleaner` class iterable and doing some crazy things to make it work but somehow I got this to work and the logic is so cyclical that it hurts to think about it still.

Reply:
Haha, no sweat, man. That's why we code review :)